### PR TITLE
Create simple and advanced views for supervisor portal

### PR DESCRIPTION
**Pull request by WackyWeaver**
## Issue
Fixes issue #441 

We created a new supervisor portal, and that is great, but we took it from a user friendly (albeit a very crappily designed) experience to a big table with a ton of info. We should give them the option to view in a clearer "simpler" view while maintaining all the same functionality and features of the more advanced view.

## Changes
- Created a button in the top right that allows you to toggle between a simple and advanced view.
- Created a new datatable that renders when Switch To Simple View is toggled that limits the information and presents it in a slightly more user friendly way.
- Added new logic in Flask to handle the rendering of a simple view.
- Modified the cookie handling logic so when the data is pulled in from the session cookies it renders the correct view.

## To Test
- Checkout `441simple_view`
- Run `./database/reset_database.sh from-backup` and `export FLASK_ENV=staging`
- `flask run`
- Upon load you should see the supervisor portal.
- Execute a search and verify that it displays properly on the simple view.
- Try sorting and make sure you can only sort by columns and fields that are directly visible to the user.
- Click "Switch to Advanced View" and verify that the data is the same but just displayed differently.
- Try sorting and make sure you can sort by all columns and fields
- Execute a search with advanced view and close the window
- Reload the window and make sure that all data is automatically loaded in properly.

### CODE REVIEW

**Review by Karina-Agliullova — APPROVED** [review]
Looks good!

File: app/static/js/supervisorPortal.js [hunk 1]
```diff
@@ -70,8 +89,14 @@ $(document).ready(function () {
   });
   $('#columnPicker').on('change', function () {
     let column = $('#columnPicker :selected').text()
-    let fields = columnFieldMap[column]
+    buttonVal = $("#switchViewButton").val()
+    let fields
+    if (buttonVal == "advanced") {
+      fields = advancedColumnFieldMap[column]
```
Comment (Mbeweg) [inline app/static/js/supervisorPortal.js]: Make this variable declaration and if statement a one-liner using a ternary operator
`let fields = buttonVal == "advanced"? advancedColumnFieldMap[column]:simpleColumnFieldMap[column];`

File: app/static/js/supervisorPortal.js [hunk 2]
```diff
@@ -168,6 +202,9 @@ function runFormSearchQuery(button) {
   }
 
   queryDict = {
+    // this may seem counterintuitive but since the value
+    // switches on button press it is technically the opposite of the current view
```
Comment (Mbeweg) [inline app/static/js/supervisorPortal.js]: These comments are out of date. The value of the switchViewButton **is equal to** the current view in the most recent commit. Should probably remove or update.

**Review by Mbeweg — COMMENTED** [review]
Changes Requested

File: app/static/js/supervisorPortal.js [hunk 3]
```diff
@@ -182,14 +214,83 @@ function runFormSearchQuery(button) {
...
+    $('#switchViewButton').html('Switch To Simple View')
+    $('#columnPicker').empty();
+    let columns = Object.keys(advancedColumnFieldMap)
+    columns.forEach((column) => {
+      var option = $('<option>', {
+        value: advancedColumnFieldMap[column][0][1],
+        text: column
+      });
+      $('#columnPicker').append(option)
+    })
```
Comment (BrianRamsay) [inline app/static/js/supervisorPortal.js]: Let's pull out 240-248 into a resetColumnPicker function (or some other name), as it is basically identical to the code in the other branch (228-236). That will make each side of the conditional look a lot simpler, and we won't repeat the code.

**Review by BrianRamsay — CHANGES_REQUESTED** [review]
Just one small change

**Review by BrianRamsay — APPROVED** [review]

**Pull request merged**